crates/services/p2p/src/service.rs

```diff
@@ -222,6 +232,9 @@ where
                     Some(TaskRequest::RespondWithGossipsubMessageReport((message, acceptance))) => {
                         report_message(&mut self.p2p_service, message, acceptance);
                     }
+                    Some(TaskRequest::RespondWithPeerReport { peer_id, peer_score, reporting_service }) => {
```

Comment:
It would be cool to make gossiping score independent. The current implementation updates the score and decides whether we want to propagate the transaction through gossiping or not by this report. But it would be nice to make those two processes independent because the gossiping score should be related only to the validity of messages and meshing.

Reply:
`RespondWithGossipsubMessageReport` deals with gossipsub,
`RespondWithPeerReport ` deals with application specific score (ie everything else except gossipsub)